dimagi/utils/couch/bulk.py

```diff
@@ -100,3 +100,33 @@ def wrapped_docs(cls, keys):
     rows = get_docs(cls.get_db(), keys)
     for row in rows:
         yield cls.wrap(row['doc'])
+
+
+class IterativeSaver(object):
+    """
+    Bulk save doc dicts in chunks.  Works well with the output from iter_docs.
+
+        with IterativeSaver(db) as iter_db:
+            for doc in iter_docs(db)
+                iter_db.save(doc)
+    """
+    def __init__(self, database, chunksize=100):
+        self.db = database
+        self.chunksize = chunksize
+
+    def __enter__(self):
+        self.to_save = []
```

Comment:
I don't know how fancy we want to get, but it could be nice to make `to_save` a dictionary, so if you try to save the same doc twice it's smart enough to only do it once